src/com/oltpbenchmark/api/BenchmarkModule.java

```
@@ -95,28 +99,37 @@ public BenchmarkModule(String benchmarkName, WorkloadConfiguration workConf, boo
         this.catalog = (withCatalog ? new Catalog(this) : null);
         File xmlFile = this.getSQLDialect();
         this.dialects = new StatementDialects(this.workConf.getDBType(), xmlFile);
+
+        try {
+            createDataSource();
+        } catch (Exception e) {
+            LOG.error("Failed to create Data source", e);
+        }
```

Why are we just logging an exception and proceeding here? Wouldn't it be better to let the workload crash at this point? (E.g. wrap and re-throw the exception as a RuntimeException if necessary.)

---

Done.